### DISCUSSION TITLE
Get the current user in a middleware

### DISCUSSION THREAD

**Discussion by leo-unglaub** [filamentphp/filament#4390] [General]
Hey,
i am having a strange problem and hope someone else can point me in the correct direction. I have registered a global middleware to all routes of my application.

```php
class Kernel extends HttpKernel
{
	protected $middleware = [
		// Laravel default entries ...
		MyMiddleware::class, // last on purpose
	];
}
```

In my middleware i check if i am in the admin panel and if yes i just want to access the current user. But even if i am logged in successfully, both `$request->user()` and `Auth::user()` are always `null`. 

Do you have any idea why? It works perfectly when i am on a normal Laravel route, but as soon as the route is from Filament, it returns null. 

Thanks and greetings
Leo

**Comment by danharrin** [comment]
Have you tried `Filament::auth()->user()` if you have a custom guard?

**Reply by leo-unglaub** [reply]
Sadly `Filament::auth()->user()` also equals to `null`. Could the problem be that my middleware gets executed before the Filament logic is fired and the user in this session gets actually set?

I have registered my middleware in `$middleware` because Filament does not use the middleware group `web`.

**Reply by leo-unglaub** [reply]
I think i found the problem, Filament does not use the middleware group `web`. So all middlewares in there are not executed. If i use the list of global middlewares for my own logic it gets executed before `Illuminate\Session\Middleware\StartSession` is executed (in the middleware group web by default) so all user stuff is not initialized yet.

**Comment by leo-unglaub** [comment]
A solution can be found her: https://github.com/filamentphp/filament/issues/4398#issuecomment-1270577064

**Comment by eighty9nine** [comment]
After checking the source code of the filament packages, on: vendor\filament\filament\src\Http\Middleware\IdentifyTenant.php, I see the get the user from the panel. I tried it, and it worked:

```php
    $panel = Filament::getCurrentPanel();
    $user = $panel->auth()->user();
```
